src/providers/microsoft/azure/mod.rs

```diff
@@ -412,7 +412,22 @@ impl MetadataProvider for Azure {
             bail!("unexpected empty certificates endpoint");
         }
 
-        let key = self.get_ssh_pubkey(certs_endpoint)?;
+        // The certs endpoint is being populated even when the user does not supply
+        // an SSH key when created a VM. In this case the certificate chain is empty
+        // causing a failure when trying to access it. Check if the certificate chain
+        // is empty, meaning that an SSH key hasn't been provided, and only warn
+        // instead of fail.
+        let key = match self.get_ssh_pubkey(certs_endpoint) {
+            Ok(k) => k,
+            Err(e) => match e.root_cause().to_string().as_ref() {
+                "SSH public key not found in chain" => {
+                    warn!("SSH pubkeys requested, but not provisioned for this instance");
```

Comment:
That sounds like a much better warning. 
While we have this open, should I change the message a few lines up as well? I was attempting to show the warning we would have expected prior to https://github.com/coreos/fedora-coreos-tracker/issues/1553

Reply:
https://github.com/coreos/afterburn/blob/f1c93ed5a6acde61f11942731766fd385f44a2e4/src/providers/microsoft/azure/mod.rs#L406

Reply:
Since that warning would display when the certs endpoint doesn't exist, maybe that one could just read: 
```
SSH pubkeys not provisioned for this instance
```

Reply:
sounds good to me